helpers/wrapper/contracts/Wrapper.sol

```diff
@@ -82,27 +81,27 @@ contract Wrapper {
     // The taker is sending ether.
     if (_takerToken == address(wethContract)) {
 
-      // Ensure the takerWallet is unset.
       require(_takerWallet == address(0),
         "TAKER_ADDRESS_MUST_BE_UNSET");
 
-      // Ensure the _takerAmount matches the ether sent.
       require(_takerAmount == msg.value,
         "VALUE_MUST_BE_SENT");
 
       // Wrap (deposit) the ether.
       wethContract.deposit.value(msg.value)();
 
-      // Approve Swap to trade it.
-      wethContract.approve(address(swapContract), msg.value);
+      // Check if Swap has allowance, if not approve Swap to trade it.
+      if (wethContract.allowance(address(this), address(swapContract)) < msg.value) {
```

Comment:
Approval is only for this transaction, think the previous approach was sufficient. Unless we'd like to add a single one-time approval in the constructor for the maximum amount, in which case we may be able to remove this logic altogether.

Reply:
will remove.